vassal-app/src/main/java/VASSAL/build/module/map/Zoomer.java

```diff
@@ -1038,8 +1038,13 @@ else if (FIT_WIDTH.equals(cmd)) {
         setZoomFactor(vd.getWidth() / md.getWidth());
       }
       else if (FIT_HEIGHT.equals(cmd)) {
-        final Dimension vd = map.getView().getVisibleRect().getSize();
-        final Dimension md = map.mapSize();
+        // A single pass (see WIDTH above) results in a poor fit for HEIGHT (ref issue #13635)
+        // Repeating the sequence corrects the issue. FIXME: This is a workaround
+        Dimension vd = map.getView().getVisibleRect().getSize();
+        Dimension md = map.mapSize();
+        setZoomFactor(vd.getHeight() / md.getHeight());
+        vd = map.getView().getVisibleRect().getSize();
+        md = map.mapSize();
```

Comment:
Sounds like it is a Java implementation issue. This can't cause any harm to other users and it is a rarely used option (that I didn't even realise existed!).